libraries/stdlib/js/src/kotlin/text/regex.kt

```diff
@@ -178,8 +178,7 @@ public actual class Regex actual constructor(pattern: String, options: Set<Regex
      * replacement for that match.
      */
     public actual fun replace(input: CharSequence, transform: (MatchResult) -> CharSequence): String {
-        var match = find(input)
-        if (match == null) return input.toString()
+        var match: MatchResult? = find(input) ?: return input.toString()
```

Comment:
Minor: usually `?: return` or `?: throw` or other non-local control flow constructions are used to ensure that the value after the flow continues normally is not null. Here this feature in fact harms more, because we have to introduce an explicit type to retain its nullability.

Reply:
Hi @ilya-g, I made this change to let the code here to be consistent with the one in jvm, if you prefer not to, I can revert this change